### ISSUE TITLE
Is websocketd compatible for HTTP/2

### ISSUE THREAD

**Issue by ljfreelancer88** [joewalnes/websocketd#387]
Hi, newbie here. Just a quick question. 

- What are the advantages of using websocketd since HTTP/2 has now the server push?
- Is websocketd compatible for HTTP/2?
- Does websocketd has a capability to "reconnect' like SSE?

Thank you in advance.

**Comment by RichardTMiles** [comment]
This weekend question really had me second guess myself, but none the less I can weigh in here. H2 allows you to send predefined resources to the browser on the initial request, or specific requests immediately. This is before the browser has time to parse the html and actually request the data. Websockts allow you to open a socket, which is a foundational OS level concept, and maintain that connection for the (optionally) life of the request. This is useful for things like instant messaging.

**Comment by asergeyev** [comment]
Websockets allow a lot more but if you can make your application work without them it's good choice. HTTP push is not duplex client/server communication but pretty good for lots of things. It also allows connection to break where re-establishing of a socket takes some effort.

**Comment by RichardTMiles** [comment]
[stackoverflow](https://stackoverflow.com/questions/28582935/does-http-2-make-websockets-obsolete)

I found myst's answer pretty informative. I agree web sockets should be used only when necessary. I've found the transport pretty slow even with gzip and other compression. In select implementations I choose only to send uri's in the payload to which the client will then request from the server using http(s). It seems counterintuitive and adds overhead, but did gain in additional security and speed. This was years back so the metrics may have shifted, but something to think about.
